src/sentry/api/endpoints/organization_details.py

```diff
@@ -248,6 +297,55 @@ def validate(self, attrs):
                 )
         return attrs
 
+    def save_trusted_relays(self, incoming, changed_data, organization):
+        timestamp_now = (
+            datetime.utcnow().isoformat() + "Z"
+        )  # make it utc relative with set timezone
+        option_key = "sentry:trusted-relays"
+        try:
+            # get what we already have
+            existing = OrganizationOption.objects.get(organization=organization, key=option_key)
+            if existing is not None:
+                key_dict = {val.get("public_key"): val for val in existing.value}
+            else:
+                key_dict = {}
+        except OrganizationOption.DoesNotExist:
+            key_dict = {}  # we don't have anything set
+            existing = None
+
+        modified = False
+        for option in incoming:
+            public_key = option.get("public_key")
+            existing_info = key_dict.get(public_key, {})
+
+            option["created"] = existing_info.get("created", timestamp_now)
+            option["last_modified"] = existing_info.get("last_modified")
+
+            # check if we modified the current public_key info and update last_modified if we did
+            if (
+                existing_info.get("public_key") is None
```

Comment:
```suggestion
                not existing_info
```

Reply:
not sure what you mean... suggestion is not equivalent... I'm checking that I have a public_key key in the existing_info dictionary... `not exsiting_info` will check against `exsiting_info` itself being `None` or empty

Reply:
Yes, but there is no way that an existing info has no `public_key`. And I thought that it might be more straight forward to just check if there was an existing info (which you coerce into an empty `dict`; it can never be `None` either), which is what the logic here suggests.

Reply:
the logic checks that public_key is not `None` which will only happen (as you point out) when we don't find the key in the existing info. 

Any way, I changed it to `not exsisting_info` as per suggestion.